modules/c++/re/tests/benchmark_regex.cpp

```diff
@@ -0,0 +1,96 @@
+
+#include <string>
+#include <fstream>
+
+#include <import/re.h>
+#include <sys/StopWatch.h>
+#include <sys/LocalDateTime.h>
+
+// First just benchmark the Regex creation time (including compile)
+double BM_RegexCreation(uint64_t numIterations)
+{
+    sys::RealTimeStopWatch sw;
+
+    sw.start();
+    for (uint64_t ii = 0; ii < numIterations; ++ii)
+    {
+        re::Regex regex("beam(Id|String)");
+    }
+    double elapsedTimeMS = sw.stop();
+
+    return elapsedTimeMS / numIterations;
+}
+
+
+// Now benchmark the actual string-matching
+double BM_RegexMatch(uint64_t numIterations, const std::string& fileString)
+{
+    sys::RealTimeStopWatch sw;
+
+    // should match multiple times
+    re::Regex regex("beam(Id|String)");
+
+    // should never match
+    //re::Regex regex("foobarfighters");
+
+    sw.start();
+    for (uint64_t ii = 0; ii < numIterations; ++ii)
+    {
+        regex.matches( fileString );
+    }
+    double elapsedTimeMS = sw.stop();
+
+    return elapsedTimeMS / numIterations;
+}
+
+
+int main(int argc, char** argv)
+{
+    if (argc < 3)
+    {
+        std::cerr << "Usage: ./bm_regex inputFile numIterations" << std::endl;
+        return 1;
+    }
+
+    uint64_t numIterations = std::stoi(argv[2]);
```

Comment:
We live in a sad world where C++11 is no guarantee.  Use `str::toType()` rather than `std::stoi()`.  Also, to protect against really lame old versions of Visual Studio that don't have `stdint.h`, rather than using `uint64_t` directly, use `sys::Uint64_T`.